sycl/source/detail/scheduler/commands.cpp

```diff
@@ -300,9 +300,12 @@ bool Command::isHostTask() const {
 }
 
 bool Command::isFusable() const {
-  return (MType == CommandType::RUN_CG) &&
-         ((static_cast<const ExecCGCommand *>(this))->getCG().getType() ==
-          CG::CGTYPE::Kernel);
+  if ((MType != CommandType::RUN_CG)) {
+    return false;
+  }
+  const auto &CG = (dynamic_cast<const ExecCGCommand &>(*this)).getCG();
```

Comment:
This can be a `static_cast` as you're checking the command type in the `if` statement above. Omit the warning and trust the `static_cast`